src/lio/os/file.c

```diff
@@ -1128,7 +1128,7 @@ int va_timestamp_set_attr(lio_os_virtual_attr_t *va, lio_object_service_fn_t *os
     n = (int)(long)va->priv;  //** HACKERY ** to get the attribute prefix length
     key = &(fullkey[n+1]);
 
-    if (strlen(fullkey) < n) {  //** Nothing to do so return;
+    if ((int)strlen(fullkey) < n) {  //** Nothing to do so return;
```

Comment:
I really hate unsigned variables of any type. I've been burned by these 
conversion issues more than I can count. As a result I only rarely use 
unsigned variables.  Making the change here is fine but I would require 
the name be changed from n ->n_unsigned or something similar.  Otherwise 
a wonderfully generic name like "n" gets used to store negative numbers 
causing all sorts of chaos.

The double conversion is because "n" is an "int" which isn't always the 
same as a "long" and leads to another warning.  The whole purpose of the 
weirdness is to keep from having to do a malloc just to store an integer.